### PR TITLE
AI: Tamed wolf defends owner

### LINKED ISSUE #2854: Tamed wolf doesn't protect owner
"*Tamed wolves will attack players or mobs that injure their owner*" (http://minecraft.gamepedia.com/Dog) They don't protect their owner. Is it possible to fix that?

### PR DESCRIPTION
**Pull request by LogicParrot**
Closes #2854.

### CODE REVIEW

File: src/Mobs/Wolf.cpp [hunk 1]
```diff
@@ -29,7 +29,7 @@ cWolf::cWolf(void) :
 
 bool cWolf::DoTakeDamage(TakeDamageInfo & a_TDI)
 {
-	if (super::DoTakeDamage(a_TDI))
+	if (!super::DoTakeDamage(a_TDI))
```
Comment (LogicParrot) [inline src/Mobs/Wolf.cpp]: Regardless of this change, the `!` should have been there previously. Is there a reason it's not, or is it just a bug?

File: src/Mobs/Wolf.h [hunk 2]
```diff
@@ -45,6 +45,12 @@ class cWolf :
 		m_OwnerUUID = a_NewOwnerUUID;
 	}
 
+	//
```
Comment (madmaxoft) [inline src/Mobs/Wolf.h]: Add doxycomments here - assumptions etc.

Comment (LogicParrot) [inline src/Mobs/Wolf.h]: Done

**Comment by LogicParrot** [comment]
The latest commit makes the wolf also react when the player attacks a mob, either by melee attacks or by projectile attacks.

File: src/ClientHandle.cpp [hunk 3]
```diff
@@ -1630,21 +1638,15 @@ void cClientHandle::HandleUseEntity(UInt32 a_TargetEntityID, bool a_IsLeftClick)
 					return true;
 				}
 			}
-			a_Entity->TakeDamage(*m_Attacker);
-			return false;
+			a_Entity->TakeDamage(*m_Me);
+			m_Me->AddFoodExhaustion(0.3);
```
Comment (LogicParrot) [inline src/ClientHandle.cpp]: By the way, previously this line never executed and there was no food exaustion when attacking mobs.

**Comment by LogicParrot** [comment]
Melee attacks are mainly handled by `ClientHandle::HandleUseEntity`
Projectile attacks are mainly handled by `ProjectileEntity::OnHitEntity`

**Comment by LogicParrot** [comment]
Rebased.

**Comment by madmaxoft** [comment]
Looks good to me.

**Pull request merged**